extension/src/pipeline/quickPick.ts

```diff
@@ -79,26 +142,31 @@ const pickFields = async (
     title: Title.SELECT_PLOT_X_METRIC
   })) as { file: string; key: string }[] | undefined
 
-  if (!xValues || xValues.length === 0) {
+  const xFieldInfo = verifyXFields(xValues)
+
+  if (!xFieldInfo) {
     return
   }
 
+  const { x, firstXKey } = xFieldInfo
+
   const yValues = (await quickPickUserOrderedValues(
-    items.filter(item => xValues.every(val => !isEqual(val, item.value))),
+    items.filter(item => xValues?.every(val => !isEqual(val, item.value))),
     { title: Title.SELECT_PLOT_Y_METRIC }
   )) as { file: string; key: string }[] | undefined
 
-  if (!yValues || yValues.length === 0) {
+  const yFieldInfo = verifyYFields(yValues, xValues?.length || 0)
+
+  if (!yFieldInfo) {
     return
   }
 
+  const { y, firstYKey } = yFieldInfo
+
   return {
-    fields: {
-      x: formatFieldQuickPickValues(xValues),
-      y: formatFieldQuickPickValues(yValues)
-    },
-    firstXKey: xValues[0].key,
-    firstYKey: yValues[0].key
+    fields: { x, y },
+    firstXKey,
+    firstYKey
```

Comment:
[Q] Do we really need to return `firstXKey` and `firstYKey` from this function? Could we simplify the type and extract the first keys in the caller?